### PR TITLE
smplify existing pytest reset_env_vars

### PR DESCRIPTION
**Pull request by chrinorse**
**How to categorize this PR?**

/kind enhancement
/area os
/os garden-linux

**What this PR does / why we need it**:
smplify existing pytest reset_env_vars
**Which issue(s) this PR fixes**:
Fixes #874 

**Special notes for your reviewer**:

**Release note**:

```feature user

```

### CODE REVIEW

File: tests/helper/tests/reset_env_vars.py
Comment (gyptazy) [inline tests/helper/tests/reset_env_vars.py:4]: I'd keep this on hold until #892 got merged and use a more Pythonic way (by using https://github.com/gardenlinux/gardenlinux/pull/892/files#diff-5ed4b018bb02161b990ad7d516a52d2695a738ed9968a11b86730edac180e888 `[key_val_in_file.py]`).

**Review by gyptazy — CHANGES_REQUESTED** [review]

**Review by nanory — APPROVED** [review]

**Pull request merged**
